src/open_inwoner/openzaak/clients.py

```diff
@@ -1,28 +1,565 @@
+import base64
 import logging
-from typing import Optional
+from datetime import date
+from typing import List, Optional
+
+from django.conf import settings
+from django.core.files.uploadedfile import InMemoryUploadedFile
+from django.utils.functional import SimpleLazyObject
 
 from ape_pie.client import APIClient
+from requests import HTTPError, RequestException, Response
+from zgw_consumers.api_models.base import factory
+from zgw_consumers.api_models.catalogi import Catalogus
+from zgw_consumers.api_models.constants import RolOmschrijving
 from zgw_consumers.client import build_client as _build_client
+from zgw_consumers.service import pagination_helper
 
-from open_inwoner.utils.api import JSONParserClient
+from open_inwoner.openzaak.api_models import InformatieObject
+from open_inwoner.openzaak.models import OpenZaakConfig
+from open_inwoner.utils.api import ClientError, get_json_response
 
+from ..utils.decorators import cache as cache_result
+from .api_models import (
+    InformatieObjectType,
+    OpenSubmission,
+    Resultaat,
+    ResultaatType,
+    Rol,
+    Status,
+    StatusType,
+    Zaak,
+    ZaakInformatieObject,
+    ZaakType,
+)
 from .models import OpenZaakConfig
 
+CRS_HEADERS = {"Content-Crs": "EPSG:4326", "Accept-Crs": "EPSG:4326"}
+
 logger = logging.getLogger(__name__)
 
 
+class ZakenClient(APIClient):
+    @cache_result(
+        "cases:{user_bsn}:{max_requests}:{identificatie}",
+        timeout=settings.CACHE_ZGW_ZAKEN_TIMEOUT,
+    )
+    def fetch_cases(
+        self,
+        user_bsn: str,
+        max_requests: Optional[int] = 1,
+        identificatie: Optional[str] = None,
+    ) -> List[Zaak]:
+        """
+        retrieve cases for particular user with allowed confidentiality level
+
+        :param:max_requests - used to limit the number of requests to list_zaken resource.
+        :param:identificatie - used to filter the cases by a specific identification
+        """
+        config = OpenZaakConfig.get_solo()
+
+        params = {
+            "rol__betrokkeneIdentificatie__natuurlijkPersoon__inpBsn": user_bsn,
+            "maximaleVertrouwelijkheidaanduiding": config.zaak_max_confidentiality,
+        }
+        if identificatie:
+            params.update({"identificatie": identificatie})
+
+        try:
+            response = self.get(
+                "zaken",
+                params=params,
+                headers=CRS_HEADERS,
+            )
+            data = get_json_response(response)
+            # FIXME the previous implementation that used `get_paginated_results`
```

Comment:
@alextreme do you know more about this? `max_cases` was set to 100, but for OZ that is the pagination limit, which means that it never actually tried to fetch the results of the next page. Is the page size for eSuite different?

I changed `max_cases` to be `max_requests` to limit the number of requests we do when retrieving paginated results

Reply:
I've checked the situation with eSuite Groningen, paging is set to 20 on the zaken endpoint. So yes the page size can be different

Reply:
Hmm okay, do you think increasing `max_requests` for pagination to 4 would be good? Or should I bring back the option to `zgw_consumers.pagination_helper` to stop once a maximum number of results is reached?

Reply:
I guess it depends on the situation, I wouldn't mind if the pagination used in the cases listview simply uses the same page size as the ZGW API it is connected to uses but if it was limited to 100 previously then doing at most 5 requests would keep this comparable. As long as there's a decent maximum I'm okay with it

Reply:
I'll limit it to 5 requests for cases (1 by default + 4 for pagination if it exists)